src/stdx.zig

```diff
@@ -237,6 +237,55 @@ pub const log = if (builtin.is_test)
 else
     std.log;
 
+/// Compare two values by directly comparing the underlying memory.
+///
+/// Assert at compile time that this is a reasonable thing to do for a given `T`. That is, check
+/// that:
+///   - `T` doesn't have any non-deterministic padding.
+///   - `T` doesn't embed any pointers.
+pub fn eql_repr(comptime T: type, a: *const T, b: *const T) bool {
```

Comment:
This function LGTM overall, though I don't love this name. But I also can't think of anything better :shrug:

Reply:
I was also wondering about how we can improve the name, also to avoid abbreviation.

`stdx.equal` would be fine I think, because we're testing for equality in the strictest possible sense here, the actual memory representation. So any relaxed version that does less can pay the cost of a longer name.

Reply:
Is there anyway we can determine the type of a or b and comptime assert they are the same, to avoid passing in the comptime type? Perhaps there are some downsides to this.

Reply:
I really like how we're moving to `*const` style here.

Reply:
The new name is `equal_bytewise`. 

>Is there anyway we can determine the type of a or b and comptime assert they are the same, to avoid passing in the comptime type? Perhaps there are some downsides to this.

I think we could, but I chose not to. The problem with this sort of functions is that it's not clearer whethere they operate on bytes pointed to by the pointer, or on the pointer itself. 

The best example here is asBytes, which is a genuinely scary footgun: 

https://github.com/ziglang/zig/issues/2904#issuecomment-1483914130

I _think_ in this particular case we can actually be safe, because we forbid pointers altogether, but it still feels overall best to type it out in full.

Reply:
I prefer `equal` (for the reasons above) or even `equal_bytes` to `equal_bytewise` (less syllables, shorter).

Reply:
Agreed on not dropping the type, to be explicit.